src/average.ts

```diff
@@ -0,0 +1,48 @@
+import { peek, pipe, sum } from ".";
+import ReturnValueType from "./types/ReturnValueType";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+
+/**
+ * Returns the average of the given (Iterable/AsyncIterable)<number> (mean)
+ *
+ * @example
+ * ```ts
+ * average([1, 2, 3, 4, 5]); // 6
+ * await average(toAsync([1, 2, 3, 4, 5])); // 6
+ *
+ * // with pipe
+ * pipe(
+ *  [1, 2, 3, 4, 5],
+ *  average,
+ * ); // 6
+ * ```
+ *
+ * see {@link https://fxts.dev/docs/pipe | pipe}
+ */
+function average<T extends Iterable<number> | AsyncIterable<number>>(
+  iterable: T,
+): ReturnValueType<T> {
+  let size = 0;
+
+  if (isIterable(iterable)) {
+    return pipe(
+      iterable as Iterable<number>,
+      peek(() => size++),
+      sum,
+      (a) => (size === 0 ? NaN : a / size),
```

Comment:
I think this part is the same as the logic below. Is there any way to combine it into one?

Reply:
Do you want to create it as a function?
like this..? `const divide = (a: number) => size === 0 ? NaN : a / size;`

Reply:
I guess that @hg-pyun says something like this:
```ts
if (isIterable(iterable) || isAsyncIterable(iterable)) {
  return pipe(
    iterable as Iterable<number> | AsyncIterable<number>,
    peek(() => size++),
    sum,
    (a) => (size === 0 ? NaN : a / size),
  ) as ReturnValueType<T>;
}
```

Reply:
We can also combine after this work
I think we can merge it first and then edit everything later.